### DISCUSSION TITLE
BF-SIM: New module for simulations

### DISCUSSION THREAD

**Discussion by detlefarend** [fhswf/MLPro#524] [Ideas]
Hi colleagues, 

I propose a new module bf.sim for classes related to simulation:

- New class FctStateTransition
- Transfer of classes State, Action from RL
- Class TransferFunction
- ...

Do you have further proposals?

**Comment by laxmikantbaheti** [comment]
Hi @detlefarend , I would like to talk about, if we modularise environment to Simulator, RewardFunction, Observer, etc. as Tasks. Can be customized separately for an environment. Will be interesting for adaptive Environements.

**Reply by detlefarend** [reply]
It's an interesting approach. Worth to take a closer look...

**Reply by detlefarend** [reply]
I really like this idea more and more. If you like you could concipate an alternative implementation for the dp landscape with an own pool of tasks... Try to keep it compatible with the current public interface of class Environment.
